Image build fails with "operator-sdk build" command

## Bug Report
**What did you do?**
While following the [Getting Started](https://github.com/operator-framework/getting-started) guide, everything worked fine till the image building step.
``` shell
$ operator-sdk build quay.io/<user>/memcached-operator:v0.0.1
```

**What did you expect to see?**
Expected to see a successful building of image

**What did you see instead? Under which circumstances?**
A failure with the following terminal output
``` shell
# github.com/example-inc/memcached-operator/pkg/apis/cache/v1alpha1
pkg/apis/cache/v1alpha1/memcached_types.go:42:25: cannot use &Memcached literal (type *Memcached) as type runtime.Object in argument to SchemeBuilder.Register:
	*Memcached does not implement runtime.Object (missing DeepCopyObject method)
pkg/apis/cache/v1alpha1/memcached_types.go:42:39: cannot use &MemcachedList literal (type *MemcachedList) as type runtime.Object in argument to SchemeBuilder.Register:
	*MemcachedList does not implement runtime.Object (missing DeepCopyObject method)
Error: failed to build operator binary: failed to exec []string{"go", "build", "-o", "/home/noor/go/src/github.com/example-inc/memcached-operator/build/_output/bin/memcached-operator", "-gcflags", "all=-trimpath=/home/noor/go/src/github.com/example-inc", "-asmflags", "all=-trimpath=/home/noor/go/src/github.com/example-inc", "-mod=vendor", "github.com/example-inc/memcached-operator/cmd/manager"}: exit status 2
Usage:
  operator-sdk build <image> [flags]

Flags:
      --go-build-args string      Extra Go build arguments as one string such as "-ldflags -X=main.xyz=abc"
  -h, --help                      help for build
      --image-build-args string   Extra image build arguments as one string such as "--build-arg https_proxy=$https_proxy"
      --image-builder string      Tool to build OCI images. One of: [docker, podman, buildah] (default "docker")

Global Flags:
      --verbose   Enable verbose logging
```


**Environment**
* operator-sdk version:

operator-sdk version: "v0.15.0", commit: "21a93ca379b887ab2303b0d148a399bf205c3231"

* go version:

go version: "go1.13.5 linux/amd64"

* Kubernetes version information:

``` shell
Client Version: version.Info{Major:"1", Minor:"16", GitVersion:"v1.16.3", GitCommit:"b3cbbae08ec52a7fc73d334838e18d17e8512749", GitTreeState:"clean", BuildDate:"2019-11-13T11:23:11Z", GoVersion:"go1.12.12", Compiler:"gc", Platform:"linux/amd64"}
Server Version: version.Info{Major:"1", Minor:"16", GitVersion:"v1.16.2", GitCommit:"c97fe5036ef3df2967d086711e6c0c405941e14b", GitTreeState:"clean", BuildDate:"2019-10-15T19:09:08Z", GoVersion:"go1.12.10", Compiler:"gc", Platform:"linux/amd64"}
```

* Kubernetes cluster kind: minikube

* Are you writing your operator in ansible, helm, or go?

go

**Possible Solution**
I don't have a workaround for this.

**Additional context**
I've tried googling and found existing issues regarding the ```go mod vendor``` command. However, I've noticed that all of those issue listings were for older operator-sdk versions, and this version doesn't invoke ```go mod vendor``` when running ```operator-sdk new``` command. Therefore, those fixes were not applicable.

--- a/website/content/en/docs/faq.md
+++ b/website/content/en/docs/faq.md
@@ -123,6 +123,25 @@ func (r *ReconcileMemcached) Reconcile(request reconcile.Request) (reconcile.Res
 	}
 ```
 
+## When using the "Getting Started" guide, I see deepcopy errors and image build fails. How do I fix this?
+
+When you run the ```operator-sdk generate k8s``` command while following the [Getting Started][getting-started-guide] guide, you might see an error like this
+
+``` bash
+INFO[0000] Running deepcopy code-generation for Custom Resource group versions: [cache:[v1alpha1], ] 
+F0523 01:18:27.122034    5157 deepcopy.go:885] Hit an unsupported type invalid type for invalid type, from github.com/example-inc/memcached-operator/pkg/apis/cache/v1alpha1.Memcached
+```
+
+This is because of the ```GOROOT``` environment variable not being set. More details [here][goroot-github-issue].
+
+In order to fix this, you simply need to export the ```GOROOT``` environment variable
+
+``` bash
+export GOROOT=$(go env GOROOT)
+```
+
+This will work for the current terminal instance, to make it more permanent, consider adding this line to the ```bashrc``` file.
+
 [kube-apiserver_options]: https://kubernetes.io/docs/reference/command-line-tools-reference/kube-apiserver/#options
 [controller-runtime_faq]: https://github.com/kubernetes-sigs/controller-runtime/blob/master/FAQ.md#q-how-do-i-have-different-logic-in-my-reconciler-for-different-types-of-events-eg-create-update-delete
 [finalizer]: /docs/golang/quickstart/#handle-cleanup-on-deletion
@@ -130,3 +149,5 @@ func (r *ReconcileMemcached) Reconcile(request reconcile.Request) (reconcile.Res
 [cr-faq]:https://github.com/kubernetes-sigs/controller-runtime/blob/master/FAQ.md
 [client.Reader]:https://godoc.org/sigs.k8s.io/controller-runtime/pkg/client#Reader
 [rbac]:https://kubernetes.io/docs/reference/access-authn-authz/rbac/
+[goroot-github-issue]:https://github.com/operator-framework/operator-sdk/issues/1854#issuecomment-525132306
+[getting-started-guide]:https://github.com/operator-framework/getting-started#overview

--- a/website/content/en/docs/faq.md
+++ b/website/content/en/docs/faq.md
@@ -123,7 +123,7 @@ func (r *ReconcileMemcached) Reconcile(request reconcile.Request) (reconcile.Res
 	}
 ```
 
-## When using the "Getting Started" guide, I see deepcopy errors and image build fails. How do I fix this?
+## I see deepcopy errors and image build fails. How do I fix this?
 
 When you run the ```operator-sdk generate k8s``` command while following the [Getting Started][getting-started-guide] guide, you might see an error like this
 

--- a/website/content/en/docs/faq.md
+++ b/website/content/en/docs/faq.md
@@ -125,22 +125,22 @@ func (r *ReconcileMemcached) Reconcile(request reconcile.Request) (reconcile.Res
 
 ## I see deepcopy errors and image build fails. How do I fix this?
 
-When you run the ```operator-sdk generate k8s``` command while following the [Getting Started][getting-started-guide] guide, you might see an error like this
+When you run the ```operator-sdk generate k8s``` command, you might see an error like this
 
 ``` bash
 INFO[0000] Running deepcopy code-generation for Custom Resource group versions: [cache:[v1alpha1], ] 
 F0523 01:18:27.122034    5157 deepcopy.go:885] Hit an unsupported type invalid type for invalid type, from github.com/example-inc/memcached-operator/pkg/apis/cache/v1alpha1.Memcached
 ```
 
-This is because of the ```GOROOT``` environment variable not being set. More details [here][goroot-github-issue].
+This is because of the `GOROOT` environment variable not being set. More details [here][goroot-github-issue].
 
-In order to fix this, you simply need to export the ```GOROOT``` environment variable
+In order to fix this, you simply need to export the `GOROOT` environment variable
 
-``` bash
-export GOROOT=$(go env GOROOT)
+```
+$ export GOROOT=$(go env GOROOT)
 ```
 
-This will work for the current terminal instance, to make it more permanent, consider adding this line to the ```bashrc``` file.
+This will work for the current environment. To persist this fix, add the above line to your environment's config file, ex. `bashrc` file.
 
 [kube-apiserver_options]: https://kubernetes.io/docs/reference/command-line-tools-reference/kube-apiserver/#options
 [controller-runtime_faq]: https://github.com/kubernetes-sigs/controller-runtime/blob/master/FAQ.md#q-how-do-i-have-different-logic-in-my-reconciler-for-different-types-of-events-eg-create-update-delete
@@ -150,4 +150,3 @@ This will work for the current terminal instance, to make it more permanent, con
 [client.Reader]:https://godoc.org/sigs.k8s.io/controller-runtime/pkg/client#Reader
 [rbac]:https://kubernetes.io/docs/reference/access-authn-authz/rbac/
 [goroot-github-issue]:https://github.com/operator-framework/operator-sdk/issues/1854#issuecomment-525132306
-[getting-started-guide]:https://github.com/operator-framework/getting-started#overview

